Responsive FrontEnd

marcpursals opened the pull request:
Hi, @knugie as talked by mail here you have my approach for a new frontEnd. 

To be awared:
1. I didn't test textAreas (we don't use arrays on our locales... honestly I forgot it)
2. If the amount of keys or categories is really big and the browser is slow on renderization it might be interesting include an infinite scroll on it, I have several posible solutions on it. Just ping me if you need it and I'll do my best.
3. I have a workaround to include this gem as standalone on an ubuntu distribution with nginx and be able to use it remotely with a locales git submodule. if you are interested I can document it so you can include it on your readme file.

Best regards from Barcelona.

Marc.

marcpursals committed:
initial refactor of layout.erb

marcpursals committed:
added new category layout

marcpursals committed:
added first aproximation to translations layout

marcpursals committed:
added new version name and contact info on gem

marcpursals committed:
finished first aproximation to new responsive front end

wteuber commented:
Thanks a lot for you PR, @marcpursals! I'll have a detailed look at this PR later this week.
Here are 2 early requests.
1. Gem files should not be part of the code base. Could you remove `i18n_yaml_editor-2.2.0.gem` ?
2. You added marc@marcpursals.com to the gem's email. Are you willing to support all parts of i18n_yaml_editor or just specific ones, e.g. the bootstrap-based frontend? I'm happy adding you there, just want to understand your intentions.

marcpursals commented:
Hi there @wteuber, sorry for the delay.
1. Yep, you're right. Sorry my mistake I'll remove it.
2. In the beginning I just added there in case you want to have my contact but I can help on whatever you want, frontend and gem itself. Whatever you need just ping me.

wteuber commented:
Sounds great. For smaller projects, we don't have strict contribution guidelines. I'll double-check Sage's open source policies to allow gem owners outside the organisation.

marcpursals committed:
deleted gem file

marcpursals committed:
deleted unnecesary layout html

wteuber commented:
I had a chance to look at the changes. Most of it looks really great and is fast too, great work!
Also, text fields for arrays work just like expected. But there are two things I think need fixing:
## 1. Broken layout for boolean values
![image](https://user-images.githubusercontent.com/2951339/44992043-676dbb80-af96-11e8-85de-788988ad4099.png)

----

## 2. Alignment of locales of different lengths, e.g. "fr" and "fr-CA" (I'd a such a case to the example files)
![image](https://user-images.githubusercontent.com/2951339/44991937-0f36b980-af96-11e8-90e5-6fd49104f1ea.png)


----
## A couple of minor things could be improved but are not worth blocking this PR

## The action button to apply filters should To apply filters, the button should The button should say "Apply"
![image](https://user-images.githubusercontent.com/2951339/44990902-d812d900-af92-11e8-9575-0fc60779de00.png)

----

## Header overlays content the reducing window size without being able to scroll up
![image](https://user-images.githubusercontent.com/2951339/44991027-2d4eea80-af93-11e8-803e-a6c13adcbe43.png)

----

## Spacing of blocks with different height
![image](https://user-images.githubusercontent.com/2951339/44991723-4b1d4f00-af95-11e8-8676-f300a35e2c6b.png)

marcpursals commented:
Hi Wolfgang, first of all thanks for your detailed feedback and I’m happy that we are close to get something that would fit everyones needs. 

Ok!! I’ll try this weekend to fix it!! I’ll let you know.

Best regards from Barcelona.

Marc.

----------
Marc Pursals i Garcia
Software Engineer || Telecom Engineer
637 757 338
marc@marcpursals.com
linkedin: https://www.linkedin.com/in/marcpursals/



> On 3 Sep 2018, at 16:29, Wolfgang Teuber <notifications@github.com> wrote:
> 
> I had a chance to look at the changes. Most of it looks really great and is fast too, great work!
> Also, text fields for arrays work just like expected. But there are two things I think need fixing:
> 
> 1. Broken layout for boolean values
> 
>  <https://user-images.githubusercontent.com/2951339/44992043-676dbb80-af96-11e8-85de-788988ad4099.png>
> 2. Alignment of locales of different lengths, e.g. "fr" and "fr-CA" (I'll a such a case to the example files)
> 
>  <https://user-images.githubusercontent.com/2951339/44991937-0f36b980-af96-11e8-90e5-6fd49104f1ea.png>
> A couple of minor things could be improved but are not worth blocking this PR
> 
> The action button to apply filters should To apply filters, the button should The button should say "Apply"
> 
>  <https://user-images.githubusercontent.com/2951339/44990902-d812d900-af92-11e8-9575-0fc60779de00.png>
> Header overlays content the reducing window size without being able to scroll up
> 
>  <https://user-images.githubusercontent.com/2951339/44991027-2d4eea80-af93-11e8-803e-a6c13adcbe43.png>
> Spacing of blocks with different height
> 
>  <https://user-images.githubusercontent.com/2951339/44991723-4b1d4f00-af95-11e8-8676-f300a35e2c6b.png>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/Sage/i18n_yaml_editor/pull/33#issuecomment-418131812>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AEE7vp8vcdwOl0dApmcZO8e8lriNc20sks5uXTzUgaJpZM4WNb79>.
>

wteuber commented:
Hi @marcpursals, did you have a chance addressing comments?

marcpursals commented:
Hi Wolfgang. Sorry I didn’t had time last two weeks to work on it and next two weeks it will be quite difficult. Big critical changes had been done this past weeks regarding a huge bug found on my company. I’ll try to do my best. I’ll let you know when everything will be fixed. 

Sorry again for letting you on dark.

Best regards from Barcelona.

> On 27 Sep 2018, at 23:44, Wolfgang Teuber <notifications@github.com> wrote:
> 
> Hi @marcpursals <https://github.com/marcpursals>, did you have a chance addressing comments?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/Sage/i18n_yaml_editor/pull/33#issuecomment-425252882>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AEE7vgdOfUT-44zEsn74oWGY2ICeYjTeks5ufUa4gaJpZM4WNb79>.
>
